base/src/main/java/com/smartdevicelink/proxy/rpc/KeyboardCapabilities.java

```diff
@@ -0,0 +1,184 @@
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the SmartDeviceLink Consortium Inc. nor the names of
+ * its contributors may be used to endorse or promote products derived
+ * from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.smartdevicelink.proxy.rpc;
+
+import com.smartdevicelink.proxy.RPCStruct;
+import com.smartdevicelink.proxy.rpc.enums.KeyboardLayout;
+
+import java.util.Hashtable;
+import java.util.List;
+
+/**
+ *
+ * <p><b>Parameter List</b></p>
+ *
+ * <table border="1" rules="all">
+ *  <tr>
+ *      <th>Param Name</th>
+ *      <th>Type</th>
+ *      <th>Description</th>
+ *      <th>Required</th>
+ *      <th>Notes</th>
+ *      <th>Version Available</th>
+ *  </tr>
+ *  <tr>
+ *      <td>maskInputCharactersSupported</td>
+ *      <td>Boolean</td>
+ *      <td>Availability of capability to mask input characters using keyboard. True: Available,False: Not Available</td>
+ *      <td>N</td>
+ *      <td></td>
+ *      <td>
+ *         @since SmartDeviceLink 7.1.0
+ *      </td>
+ *  </tr>
+ *  <tr>
+ *      <td>supportedKeyboardLayouts</td>
+ *      <td>List<KeyboardLayout></td>
+ *      <td>Supported keyboard layouts by HMI.</td>
+ *      <td>N</td>
+ *      <td>{"array_min_size": 1, "array_max_size": 1000}</td>
+ *      <td>
+ *         @since SmartDeviceLink 7.1.0
+ *      </td>
+ *  </tr>
+ *  <tr>
+ *      <td>configurableKeys</td>
+ *      <td>List<ConfigurableKeyboards></td>
+ *      <td>Get Number of Keys for Special characters, App can customize as per their needs.</td>
+ *      <td>N</td>
+ *      <td>{"array_min_size": 1, "array_max_size": 1000}</td>
+ *      <td>
+ *         @since SmartDeviceLink 7.1.0
+ *      </td>
+ *  </tr>
+ * </table>
+ * @since SmartDeviceLink 7.1.0
+ */
+public class KeyboardCapabilities extends RPCStruct {
+    /**
```

Comment:
@kostyaBoss I would remove lines 87-89 since it is not part of the proposal's XML.

Reply:
@santanamk, This file is new and introduced in the XML related. Do we still need to rollback (these changes are from the generator)

Reply:
@kostyaBoss So I went through the `KeyboardCapabilities.java` file and the only differences I see is (when comparing my generated file with yours) is in a few places you have added the `@ s i n c e SmartDeviceLink 7.1.0` where the generator didn't generate it.  I think this is okay.  You don't need to change anything.